src/linear_algebra/general_linear_group.lean

```diff
@@ -84,6 +89,17 @@ begin
   exact inv_of_eq_nonsing_inv (↑A : matrix n n R),
 end
 
+/-- An element of the matrix general linear group on `(n) [fintype n]` can be considered as an
+element of the endomorphism general linear group on `n → R`. -/
+def to_linear : general_linear_group n R ≃* linear_map.general_linear_group R (n → R) :=
+units.map_equiv matrix.to_lin_alg_equiv'.to_ring_equiv.to_mul_equiv
+
+-- TODO This simp-lemma should be stated for `n` rather than `fin m`, but for some reason it
+-- doesn't trigger when stated in the larger generality.  Decidability issue apparently?
+@[simp] lemma coe_to_linear {m : ℕ} (M : general_linear_group (fin m) R) :
+  (to_linear M : (fin m → R) →ₗ[R] (fin m → R)) = matrix.mul_vec_lin M :=
+rfl
```

Comment:
If this is the case, you should provide both versions of the lemma; even if it mysteriously doesn't work for `simp`, it's still useful to have the lemma for `rw`.